providers/azureblob/src/main/java/org/jclouds/azure/storage/filters/SharedKeyLiteAuthentication.java

```diff
@@ -81,27 +87,86 @@
    @Inject
    public SharedKeyLiteAuthentication(SignatureWire signatureWire,
          @org.jclouds.location.Provider Supplier<Credentials> creds, @TimeStamp Provider<String> timeStampProvider,
-         Crypto crypto, HttpUtils utils) {
+         Crypto crypto, HttpUtils utils, @Named("sasAuth") boolean sasAuthentication) {
       this.crypto = crypto;
       this.utils = utils;
       this.signatureWire = signatureWire;
+	  this.storageUrl = URI.create("https://" + creds.get().identity + ".blob.core.windows.net/");
       this.creds = creds;
+	  this.identity = creds.get().identity;
+	  this.credential = creds.get().credential;
       this.timeStampProvider = timeStampProvider;
+	  this.isSAS = sasAuthentication;
    }
-
+   
+   /** 
+   * this is an updated filter method, which decides whether the SAS or SharedKeyLite 
+   * is used and applies the right filtering.  
+   * 
+   */
    public HttpRequest filter(HttpRequest request) throws HttpException {
+      if (this.isSAS){
+         request = filterSAS(request, this.credential);
+      } else {
+         request = filterKey(request);
+      }
+      utils.logRequest(signatureLog, request, "<<");
+      return request;
+   }
+   
+   /** 
+   * this filter method is applied only for the cases with SAS Authentication. 
+   * 
+   */
+   public HttpRequest filterSAS(HttpRequest request, String credential) throws HttpException {
+      String containerName = null;
+      String blobName = null;
+      URI requestUri = request.getEndpoint();
+      try {
+         String[] parametersArray = cutUri(requestUri); 
+         containerName = parametersArray[1];
+		 
+         if (parametersArray.length == 3) {
+            blobName = parametersArray[2];
+            return removeAuthorizationHeader(replaceDateHeader(request.toBuilder().endpoint(Uris.uriBuilder(storageUrl).appendPath(containerName).appendPath(blobName).query(credential).build()).build()));
```

Comment:
I meant to split that long line in something like:
```suggestion
            return removeAuthorizationHeader(
                  replaceDateHeader(request.toBuilder()
                     .endpoint(Uris.uriBuilder(storageUrl)
                        .appendPath(containerName)
                        .appendPath(blobName)
                        .query(credential)
                        .build())
                     .build()));
```